src/api/client.ts

```diff
@@ -212,6 +214,33 @@ export class WsClient extends Emittery {
         sendHandler(request, resolve, reject);
       });
       return promise as Promise<Response>;
+    } else if (this.url && this.authenticationToken) {
+      const response = new Promise<unknown>((resolve, reject) => {
+        // typescript forgets in this promise scope that this.url is undefined
```

Comment:
I know I wrote this, but it didn't make sense =D
```suggestion
        // typescript forgets in this promise scope that `this.url` is not undefined
```

Reply:
I seem to struggle with double negations :)

Reply:
You wrote it correctly and I muddled it up.